xray/backends/pynio.py

```diff
@@ -0,0 +1,247 @@
+import warnings
+
+import numpy as np
+
+from .. import Variable
+from ..conventions import pop_to, cf_encoder
+from ..core import indexing
+from ..core.utils import FrozenOrderedDict, NDArrayMixin, close_on_error
+from ..core.pycompat import iteritems, basestring, OrderedDict
+
+from .common import AbstractWritableDataStore
+from .netcdf3 import (encode_nc3_attr_value, encode_nc3_variable,
+                      maybe_convert_to_char_array)
+
+# This lookup table maps from dtype.byteorder to a readable endian
+# string used by netCDF4.
+_endian_lookup = {'=': 'native',
+                  '>': 'big',
+                  '<': 'little',
+                  '|': 'native'}
+
+
+class NetCDF4ArrayWrapper(NDArrayMixin):
+    def __init__(self, array):
+        self.array = array
+
+    @property
+    def dtype(self):
+        dtype = np.dtype(self.array.typecode())
+        if dtype is str:
+            # return object dtype because that's the only way in numpy to
+            # represent variable length strings; it also prevents automatic
+            # string concatenation via conventions.decode_cf_variable
+            dtype = np.dtype('O')
+        return dtype
+
+    def __getitem__(self, key):
+        print 'in pynio getitem', key
+        print self.array.name
```

Comment:
Need to get rid of these print-statements, or replace them with logging statements.